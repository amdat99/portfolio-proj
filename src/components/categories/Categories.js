import React,{useEffect,useState} from 'react';
import { withRouter } from 'react-router-dom'
import { connect } from 'react-redux';
import { isCategoryDataLoaded,selectCategory } from '../../redux/shop/shop.selectors'
 import ItemsCollection from '../items-collection/Items-collection'


 import ShopHeader from '../../components/shop-header/Shop-header'
 
 import { createStructuredSelector } from 'reselect';
 import {  fetchCategoryPending} from '../../redux/shop/shop.actions'

 import './Categories.scss'
    export function Categories({match,isLoaded,categoryItems,fetchCategoryPending}) {
    
    const [categoryList]=useState(['clothes','shoes','electronics','food','general'])


    useEffect(()=>{
    categoryList.map( category =>{
    if( match.url === '/category/'+category ){
       return fetchCategoryPending(category)} 
       })
    },[match,fetchCategoryPending,categoryList])
     
    return (
    <div>
        <ShopHeader />
        
  
        <div>
        <h1 id="categories-title"> {match.params.categoryId}</h1>
        { isLoaded
    ?<div id='categories-container'>
        {categoryItems.map((item, i) => (
            <ItemsCollection key={i} item={item} className='collection-item'/>
          ))}
    </div>
        :<h1>loading...</h1>
        }  
        </div>      
        
    </div>
    );
}

const mapStateToProps = (createStructuredSelector)({
    categoryItems: selectCategory,
    isLoaded: isCategoryDataLoaded,
    })

  
const mapDispatchToProps = (dispatch) => ({
    fetchCategoryPending: (category) => dispatch(fetchCategoryPending(category)),
}) 

export default withRouter(connect(mapStateToProps,mapDispatchToProps)(Categories));