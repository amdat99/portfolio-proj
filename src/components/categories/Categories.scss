#categories-container{
    margin-top: 200px;
}

#categories-title{
    margin-top: 50px;
}