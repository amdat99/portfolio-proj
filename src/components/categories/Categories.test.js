import React from 'react';

import { shallow} from 'enzyme';
import {createSerializer} from 'enzyme-to-json';
import toJson from 'enzyme-to-json';

import {Categories} from './Categories'

expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));

        let wrapper
        let wrapper2
        let mockfetchCategoryPending

        const mockMatch = {
            params: 'shoes',
            url: 'shoes'
          };
      

        
  beforeEach(() => {
   
    mockfetchCategoryPending = jest.fn();

   const mockProps = {
            categoryItems: {
              picture: 'www.picture.com',
              price: 10,
              name: 'ball',
              quantity: 5,
              profileId: 123
            },
            match: mockMatch,
            fetchCategoryPending: mockfetchCategoryPending

            }
        
          
           wrapper = toJson(shallow(<Categories  {...mockProps} />))
           wrapper2 = shallow(<Categories {...mockProps} />);
        })
        



it('expect to render cartItem component', () => {
    expect(wrapper).toMatchSnapshot();
    

})

it('expect fetchcategory pending to be called on render', () => {
    const match = '/category/'+'shoes'
    if( match.url === '/category/'+'shoes'){
   expect(mockfetchCategoryPending).toHaveBeenCalled();
    }

    // it('the counrt numbet should be rendered', () => {
    //     const match = parseInt(wrapper2.find('.categories-title').text());
    //     expect(match.params).totoBe('shoes');
    //   });

      
// it('the count number should be rendered', () => {
//     const itemCount = parseInt(wrapper2.find('.cart-button-count').text());
//     expect(itemCount).toBe(1);
//   });

 
})

