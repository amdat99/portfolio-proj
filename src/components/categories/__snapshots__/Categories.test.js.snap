// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render cartItem component 1`] = `
<div>
  <Connect(ShopHeader) />
  <div>
    <h1
      id="categories-title"
    >
       
    </h1>
    <h1>
      loading...
    </h1>
  </div>
</div>
`;
